akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -30,10 +30,22 @@ abstract class GraphStageWithMaterializedValue[+S <: Shape, +M] extends Graph[S,
 
   protected def initialAttributes: Attributes = Attributes.none
 
-  final override lazy val traversalBuilder: TraversalBuilder = {
-    val attr = initialAttributes
-    TraversalBuilder.atomic(GraphStageModule(shape, attr, this), attr)
-  }
+  private[this] final var _traversalBuilder: TraversalBuilder = _
```

Comment:
I'll investigate in depth, it could be that it's safe anyway since immutable except the port id mutation that we do but that results in the same 0 being written always hm.